### DISCUSSION TITLE
middleware cookie not available in server component

### DISCUSSION THREAD

**Discussion by Arunjenson** [vercel/next.js#61818] [App Router]
cookie setted via middleware is not available cookieStore.getAll() in my server component page.jsx

**Comment by icyJoseph** [comment]
Hi,

I think you have to do something like this:

```ts
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";

type Cookie = Parameters<
  ReturnType<typeof NextResponse.next>["cookies"]["set"]
>[0];

export function middleware(request: NextRequest) {
  const cookie = { name: "vercel", value: "fast", path: "/" } satisfies Cookie;
  request.cookies.set(cookie.name, cookie.value);

  const response = NextResponse.next(request.clone());
  response.cookies.set(cookie);

  return response;
}

```

A couple of take aways:

- When you set the response cookie, that's what will be outgoing down to the client, but it is not present in the incoming request to you server. 
- In the above snippet, we apply the cookie to both the response going back to the user, but also to the request that we take in
